src/shared/components/Search/Search.tsx

```diff
@@ -0,0 +1,171 @@
+import styled from '@emotion/styled'
+import React, { useCallback, useMemo, useRef, useState } from 'react'
+
+import { AssetAvailability, SearchQuery, VideoFieldsFragment, useSearchLazyQuery } from '@/api/queries'
+import { SvgGlyphSearch } from '@/shared/icons'
+import { colors, transitions } from '@/shared/theme'
+
+import { IconButton } from '../IconButton'
+import { TextField } from '../TextField'
+
+export type SearchProps = {
+  searchInputRef: React.RefObject<HTMLInputElement>
+  isSearchInputOpen: boolean
+  setIsSearchingInputOpen: (isOpen: boolean) => void
+  setIsSearching: (isOpen: boolean) => void
+  search: (searchQuery: string) => void
+  width?: string
+  className?: string
+}
+export const Search: React.FC<SearchProps> = ({
+  searchInputRef,
+  isSearchInputOpen,
+  setIsSearching,
+  search,
+  setIsSearchingInputOpen,
+  width,
+  className,
+}) => {
+  const [searchQuery, setSearchQuery] = useState('')
+  const handleSearchInputKeyPress = useCallback(
+    (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === 'Enter' || event.key === 'NumpadEnter') {
+        if (searchQuery.trim() === '') {
+          setSearchQuery('')
+          setIsSearching(false)
+        } else {
+          search(searchQuery)
+          setIsSearching(true)
+        }
+      }
+      if (event.key === 'Escape' || event.key === 'Esc') {
+        setIsSearchingInputOpen(false)
+        searchInputRef.current?.blur()
+        setSearchQuery('')
+      }
+    },
+    [search, searchInputRef, searchQuery, setIsSearching, setIsSearchingInputOpen, setSearchQuery]
+  )
+
+  return (
+    <SearchContainer className={className}>
```

Comment:
I said this before in other PR and I still maintain this. We should just use `form` here, like this:
```tsx
    <SearchContainer
      as="form"
      className={className}
      onSubmit={(e) => {
        e.preventDefault()
        // all on submit logic like search(searchQuery)
      }}
    >
```
Hitting 'enter` will just work out of the box(but we could leave listening on `escape` though).